tools/topology/topology2/platform/intel/dmic-wov.conf

```diff
@@ -0,0 +1,211 @@
+		}
+		Object.Widget.copier.1 {
+			stream_name $DMIC1_PCM_CAPS
+			Object.Base.audio_format.1 {
+				in_rate			16000
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_rate		16000
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_rate			16000
+				in_channels		4
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_rate		16000
+				out_channels		4
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+			}
+		}
+	}
+
+	passthrough-be.12 {
+		direction	"capture"
+		format 	$FORMAT
+		rate	16000
+		rate_min	16000
+		rate_max	16000
+		index		$DMIC1_DAI_PIPELINE_ID
+		Object.Widget.pipeline.1 {
+			stream_name $DMIC1_PIPELINE_STREAM_NAME
+		}
+
+		Object.Widget.copier.1 {
+			dai_index	1
+			type		"dai_out"
+			dai_type	"DMIC"
+			copier_type	"DMIC"
+			type		dai_out
+			stream_name	$DMIC1_NAME
+			node_type $DMIC_LINK_INPUT_CLASS
+			Object.Base.audio_format.1 {
+				in_rate			16000
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_rate		16000
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_rate			16000
+				in_channels		4
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_rate		16000
+				out_channels		4
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+			}
+		}
+
+		Object.Widget.kpb.1 {
+			Object.Base.audio_format.1 {
+				in_rate			16000
+				out_rate		16000
+				obs			128
+				in_ch_cfg	$CHANNEL_CONFIG_STEREO
+				in_ch_map	$CHANNEL_MAP_STEREO
+				out_ch_cfg	$CHANNEL_CONFIG_STEREO
+				out_ch_map	$CHANNEL_MAP_STEREO
+				in_channels	2
+				out_channels	2
+				out_bit_depth	32
+				in_bit_depth	32
+				in_valid_bit_depth	32
+				out_valid_bit_depth	32
+			}
+			Object.Base.audio_format.2 {
+				in_rate			16000
+				out_rate		16000
+				obs			128
+				in_ch_cfg	$CHANNEL_CONFIG_4_POINT_0
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_4_POINT_0
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+				in_channels	4
+				out_channels	4
+				out_bit_depth	32
+				in_bit_depth	32
+				in_valid_bit_depth	32
+				out_valid_bit_depth	32
+			}
+			Object.Base.src_pin_binding.1 {
+				src_pin_binding_wname $WOV_PIPELINE_SINK
+			}
+			Object.Base.src_pin_binding.2 {
+				src_pin_binding_wname $DMIC1_HOST_PIPELINE_SINK
+			}
+			Object.Base.src_pin_binding.3 {
+				src_pin_binding_wname "NotConnected"
+			}
+		}
+	}
+	wov-detect.1 {
+		index		$WOV_PIPELINE_ID
+		format 		s32le
+		channels	2
+		Object.Widget.wov.1 {
+			uuid	$WOV_UUID
+			cpc	$WOV_CPC
```

Comment:
Can you please remind me what would happend if we dont set the cpc?

Reply:
> Can you please remind me what would happen if we dont set the cpc?

This is a good question! I tried not to set cpc before when enabling wov SOF fw. It seemed to work even without cpc setting. Maybe we need CoE expert help to answer this question. @iganakov

Reply:
And maybe we can enhance the fw to set the cpc in FW itself and don't need tplg setting it.

Reply:
was also wondering if the cpc is needed in every widget, we should probably add it to base class etc. I guess it is originally used to count the pipeline cycles so it has "room" to run, not sure what is the use in practice.

Reply:
if we dont need cpc for wov, is it needed for kpb and micsel?